### PR TITLE
feat: Add API for chain data explorer

### LINKED ISSUE #432: [FEATURE] New API for chain data explorer
## Is your feature request related to a problem? Please describe.
There seems to be a need to see block and transaction data in wallet apps. On the other hand, implementing or building a dedicated blockchain explorer is expensive.

We are also dissatisfied with the lack of OSS blockchain explorers that are suitable for use in ibet.

## Describe the solution you'd like

We would like to add an API that provides block explorer functionality.
- Detailed specifications for this API are under consideration.
- Also, consider decoding and returning transaction data that can be decoded by ABI.

### PR DESCRIPTION
**Pull request by YoshihitoAso**
close #432 

### API

- [NEW] GET: /blockchain_explorer/block_data
  - Returns a list of block data within the specified block number range. The maximum number of search results is 1000.
- [NEW] GET: /blockchain_explorer/block_data/{block_number}
  - Returns block data in the specified block number.
- [NEW] GET: /blockchain_explorer/tx_data
  - Returns a list of transactions by various search parameters. The maximum number of search results is 10000.
- [NEW] GET: /blockchain_explorer/tx_data/{hash}
  - Searching for the transaction by transaction hash.

### Indexer

- Add Indexer for indexing BlockData and TxData.
- New environment variable: BC_EXPLORER_ENABLED (default: 0)

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -112,6 +115,20 @@ async def validation_exception_handler(request: Request, exc: RequestValidationE
...
+    meta = {
+        "code": 1,
+        "title": "RequestValidationError"
+    }
+    return JSONResponse(
+        status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+        content=jsonable_encoder({"meta": meta, "detail": exc.errors()}),
+    )
+
+
```
Comment (YoshihitoAso) [inline app/main.py:131]: Added exception handler for query validation.

File: app/model/db/idx_block_data.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+from .base import Base
+
+
+class IDXBlockData(Base):
+    """Block data (INDEX)"""
+    __tablename__ = "block_data"
+
+    # Header data
+    number = Column(BigInteger, primary_key=True, autoincrement=False)
```
Comment (YoshihitoAso) [inline app/model/db/idx_block_data.py:36]: Autoincrement is explicitly turned off.

File: migrations/versions/ff1d28df9f37_v21_12_0_feature_432.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
+"""v21_12_0_feature_432
```
Comment (purplesmoke05) [inline migrations/versions/ff1d28df9f37_v21_12_0_feature_432.py:1]: `v22_12_0` would be nice.

**Review by purplesmoke05 — APPROVED** [review]
LGTM with nits

File: migrations/versions/ff1d28df9f37_v21_12_0_feature_432.py (hunk 3 above)
Comment (YoshihitoAso) [reply migrations/versions/ff1d28df9f37_v21_12_0_feature_432.py:1]: Oops. I guess I was in a bit of a time warp.

**Pull request merged**